File: client/android/app/src/main/java/com/stonefive/chalkak/feature/feed/component/FeedTopic.kt
package com.stonefive.chalkak.feature.feed.component

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import com.stonefive.chalkak.core.designsystem.theme.ChalkakTheme

@Composable
fun FeedTopic(
    dateLabel: String,
    topic: String,
    modifier: Modifier = Modifier,
) {
    Column(
        modifier = modifier
            .padding(
                start = ChalkakTheme.spacing.screenHorizontal,
                end = ChalkakTheme.spacing.screenHorizontal,
                top = 16.dp,
                bottom = 40.dp,
            ),
    ) {
        Text(
            text = dateLabel,
            color = ChalkakTheme.colors.textPrimary,
            style = ChalkakTheme.typography.subheadline,
        )
        Text(
            text = topic,
            color = ChalkakTheme.colors.textPrimary,
            style = ChalkakTheme.typography.display,
            modifier = Modifier.padding(top = 8.dp),
        )
    }
}

@Preview(showBackground = true, widthDp = 402)
@Composable
private fun FeedTopicPreview() {
    ChalkakTheme {
        FeedTopic(
            dateLabel = "8월 3일의 주제",
            topic = "하늘하늘하늘",
            modifier = Modifier.fillMaxWidth(),
        )
    }
}

File: client/iOS/chalkak/chalkak/Features/Feed/Components/FeedTopic.swift
import SwiftUI

struct FeedTopic: View {
    @Environment(\.chalkakTheme) private var theme
    let dateLabel: String
    let topic: String

    var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            Text(dateLabel)
                .font(theme.typography.subheadline)
                .foregroundStyle(theme.colors.textPrimary)

            Text(topic)
                .font(theme.typography.display)
                .foregroundStyle(theme.colors.textPrimary)
                .padding(.top, Metrics.topicTopSpacing)
        }
        .frame(maxWidth: .infinity, alignment: .leading)
    }
}

private enum Metrics {
    static let topicTopSpacing: CGFloat = 8
}

#Preview("Feed Topic", traits: .sizeThatFitsLayout) {
    FeedTopic(dateLabel: "8월 3일의 주제", topic: "하늘하늘하늘")
        .padding(.horizontal, ChalkakTheme.light.spacing.screenHorizontal)
        .background(ChalkakTheme.light.colors.background)
        .chalkakTheme(.light)
}
